--- ./php_imap.c.bak	2024-10-15 18:02:11
+++ ./php_imap.c	2026-04-27 21:07:21
@@ -32,6 +32,21 @@
 #endif
 
 #include "php.h"
+#ifndef INI_STR
+# define INI_STR(name) zend_ini_string((name), (size_t)strlen(name), 0)
+#endif
+#ifndef INI_INT
+# define INI_INT(name) zend_ini_long((name), (size_t)strlen(name), 0)
+#endif
+#ifndef INI_FLT
+# define INI_FLT(name) zend_ini_double((name), (size_t)strlen(name), 0)
+#endif
+#ifndef INI_BOOL
+# define INI_BOOL(name) zend_ini_parse_bool(zend_ini_str((name), (size_t)strlen(name), 0))
+#endif
+#ifndef EMPTY_SWITCH_DEFAULT_CASE
+# define EMPTY_SWITCH_DEFAULT_CASE() default: ZEND_UNREACHABLE();
+#endif
 #include "php_ini.h"
 #include "php_streams.h"
 #include "Zend/zend_attributes.h"
 