--- ./php_scws.c.bak	2023-02-04 12:26:21
+++ ./php_scws.c	2026-04-27 21:07:25
@@ -23,6 +23,21 @@
 #endif
 
 #include "php.h"
+#ifndef INI_STR
+# define INI_STR(name) zend_ini_string((name), (size_t)strlen(name), 0)
+#endif
+#ifndef INI_INT
+# define INI_INT(name) zend_ini_long((name), (size_t)strlen(name), 0)
+#endif
+#ifndef INI_FLT
+# define INI_FLT(name) zend_ini_double((name), (size_t)strlen(name), 0)
+#endif
+#ifndef INI_BOOL
+# define INI_BOOL(name) zend_ini_parse_bool(zend_ini_str((name), (size_t)strlen(name), 0))
+#endif
+#ifndef EMPTY_SWITCH_DEFAULT_CASE
+# define EMPTY_SWITCH_DEFAULT_CASE() default: ZEND_UNREACHABLE();
+#endif
 #include "php_ini.h"
 #include "ext/standard/info.h"
 #include "php_scws.h"
 